torchtitan/config_manager.py

```diff
@@ -332,8 +332,9 @@ def __init__(self):
             default="bfloat16",
             choices=["bfloat16", "float32"],
             help="""
-                torch dtype to use for parameters when applying mixed precision via FSDP.
-                This feature only takes effect when data_parallel_degree > 1
+                torch dtype to use for parameters when applying mixed precision.
+                When data_parallel_degree > 1, this changes FSDP's `param_dtype`.
+                When data_parallel_degree == 1, this enables AMP autocast.
```

Comment:
I'm a bit confused by the description here. Other than DDP, does it work with TP, PP, CP, etc.?
I wonder if there is a document/tutorial on this.
cc: @fegin

Reply:
the implementation of mixed precision only needs to change based on whether `fully_shard` (which has its own internal mixed prec mechanism) or `replicate` (which internally wraps a module with DDP) is used. 

therefore, the qn of what happens (and what *should* happen) w/ TP/PP/DP is solely dependent on the sharded data parallel degree.

Reply:
what happens if neither FSDP or DDP is used? btw DDP is not composable with TP/PP today as far as I know.

Reply:
If FSDP is not used, AMP should be the default answer for all other parallelisms. But we don't actually test AMP + TP.

Reply:
do you mean the behavior of dtensor TP sharded operations under AMP in general pytorch is not tested? or that specifically it is not tested in torchtitan

Reply:
DTensor TP+ AMP